pandas/core/internals/blocks.py

```diff
@@ -1306,26 +1307,34 @@ def shift(self, periods, axis: int = 0, fill_value=None):
         return [self.make_block(new_values)]
 
     def where(
-        self, other, cond, errors="raise", try_cast: bool = False, axis: int = 0,
+        self,
+        other,
+        cond: np.ndarray,
```

Comment:
this type annotation conflicts with the code below?

Reply:
yah, i went back and forth on this.  the "spirit" of the annotation/docstring-type is right here, but it misses edge cases.  i guess yah, this should be made precise.  will update.

Reply:
Is it an edge case? Because it might also be solved then?
In what cases is `cond` not an ndarray? (whenever the user passes a series with boolean dtype higher up?)

Reply:
the cases we get here with anything other than ndarray[bool] are SparseArray[bool] and BooleanArray.